docker/resource_docker_plugin.go

```diff
@@ -0,0 +1,67 @@
+package docker
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceDockerPlugin() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDockerPluginCreate,
+		Read:   resourceDockerPluginRead,
+		Update: resourceDockerPluginUpdate,
+		Delete: resourceDockerPluginDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"plugin_reference": {
+				Type:        schema.TypeString,
+				Description: "Docker Plugin Reference.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"alias": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Docker Plugin alias.",
+			},
+			"disabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"grant_all_permissions": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, grant all permissions necessary to run the plugin",
+			},
+			"args": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"disable_when_set": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, the plugin becomes disabled temporarily when the plugin setting is updated",
+			},
+			"force_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"enable_timeout": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "HTTP client timeout to enable the plugin",
+			},
+			"force_disable": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, then the plugin is disabled forcibly when the plugin is disabled.",
+			},
+		},
```

Comment:
Can you add config and settings computed properties that expose the values in the Plugin data.
You can use github.com/mitchellh/mapstructure to automatically convert the go classes to maps before calling d.Set().
See https://github.com/brinkmanlab/terraform-provider-galaxy/blob/09ce5c679ae4625e240fa98e9a46322a645fae1f/galaxy/resource_job.go#L194-L200 for an example.

Reply:
nice pointer with `mapstructure` @innovate-invent. which properties exactly do you mean to convert?

Reply:
Nothing specific, it just provides information about the plugin that are likely important to other resources. A good example is accessing the preconfigured mounts the plugin uses so that a resource that uses it can also mount the mount point.

Reply:
Ok, I'd keep this improvement for the next iteration/a separate issue.